lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -115,8 +122,70 @@ void SetMessageManagerHandler::SetAmfSessionContext(
         MLOG(MINFO)
             << "AMF request type INITIAL_REQUEST and session state CREATING";
         auto session_map = session_store_.read_sessions({imsi});
-        send_create_session(session_map, imsi, session_id, cfg, dnn);
+        send_create_session(session_map, imsi, session_id, cfg, pdu_id);
+      }
+    }
+    response_callback(Status::OK, SmContextVoid());
+  });
+}
+
+/* Handling set message from AMF
+ * check if PDU_SESSION exists
+ * if the PDU_SESSION doen't exists, log and ignore
+ * If EXISTING_PDU_SESSION, get the session entry, check on incoming session
+ * state and version  accordingly take action, write to memory by SessionStore
+ */
+
+void SetMessageManagerHandler::SetSmfNotification(
+    ServerContext* context, const SetSmNotificationContext* notif,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  auto& notif_cpy = *notif;
+  // Print the message from AMF
+  PrintGrpcMessage(static_cast<const google::protobuf::Message&>(notif_cpy));
+
+  // Requested message from AMF to release the session
+  m5g_enforcer_->get_event_base().runInEventBaseThread([this, response_callback,
+                                                        notif_cpy]() {
+    /* Read the proto message and check for state. Get the config out of proto.
+     * Code for Active state notification
+     */
+    // extract values from proto
+    auto pdu_id = notif_cpy.rat_specific_notification().pdu_session_id();
+    auto imsi       = notif_cpy.common_context().sid().id();
+    std::string dnn = notif_cpy.common_context().apn();
+    /* Read the SessionMap from global session_store */
+    SessionSearchCriteria criteria(imsi, IMSI_AND_PDUID, pdu_id);
+    auto session_map       = session_store_.read_sessions({imsi});
+    auto session_it        = session_store_.find_session(session_map, criteria);
+    std::string session_id = id_gen_.gen_session_id(imsi);
+    auto& session          = **session_it;
+    if (session_it) {
+      MLOG(MINFO) << "Received Session Notificaiton from UE with IMSI: " << imsi
+                  << " of session Id " << session->get_session_id();
+      if (notif_cpy.common_context().sm_session_version() == 0) {
+        MLOG(MERROR) << "Wrong version received from AMF for IMSI " << imsi;

```

Comment:
no error is returned on grpc response?

Reply:
Clarify how shud we send error on GRPC reponse, cud u point any earlier code ?

Reply:
Fixed,